### PR TITLE
Jakub observation cards redesign 177

### LINKED ISSUE #177: Map explorer - display more thumbnails at one time
Make more images visible at any one time to speed up tagging and reduce scrolling. Note that high resolution images are currently loaded full size and significantly slow down the app.

Here are some screenshots from a similar app that classifies animals and is then used for validation:
WhatsApp Image 2021-07-21 at 15 35 21
WhatsApp Image 2021-07-21 at 15 35 22
WhatsApp Image 2021-07-21 at 15 38 45

### PR DESCRIPTION
**Pull request by jchojna**
Closes #177

### Description

-  move observations panel to cover the map component
-  create custom observations box's header with title and back button (back to map or back to observations)
-  create grid of cards
-  change card element according to the mock-up
-  update Tooltip component
-  display information on card's hover
-  add functionality to maximize a card
-  handle 'back to observations' button
-  fix issue with filters not working when observations panel visible
-  remove drawer from ObservationsInspector
-  add simple responsiveness to observations list
-  fix issue with select's value - prediction not active in select menu
-  add overriden label
-  fix tooltip on dropdown hover
-  try to refresh cards on filters change
-  refactor code and typings

### How to test

- run the app and experiment with observations list component, new features, filtering observations and overriding predictions

### CODE REVIEW

File: app/components/ObservationsInspector.tsx [hunk 1]
```diff
@@ -1,166 +1,77 @@
...
-    return null;
-  }
+  const [maximizedCardIndex, setMaximizedCardIndex] = useState<number>(-1);
+  const [isMaximizedMode, setMaximizedMode] = useState<boolean>(false);
+  const toggleMaximizedMode = (cardIndex: number) => {
+    setMaximizedCardIndex(cardIndex);
+    setMaximizedMode(cardIndex >= 0);
+  };
+  const backButtonText = t(`explore.${isMaximizedMode ? 'backToObservations' : 'backToMap'}`);
+  const onBackButtonClick = isMaximizedMode ? () => toggleMaximizedMode(-1) : onClose;
```
Comment (kamilzyla) [inline app/components/ObservationsInspector.tsx:29]: I find this slightly easier to read, WDYT? :wink:
```suggestion
  const onBackButtonClick = () => isMaximizedMode ? setMaximizedCardIndex(null) : onClose();
```

**Review by kamilzyla — CHANGES_REQUESTED** [review]
This is a huge improvement for the observation inspector, GJ! :tada:

File: app/assets/translations.json [hunk 2]
```diff
@@ -180,6 +181,8 @@
       },
       "explore": {
         "chooseFile": "Choisissez le fichier de résultats à explorer",
+        "backToMap": "Retour à la carte",
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/assets/translations.json:184]: @kamilzyla I verified it with my girlfriend, who speaks French very well, but I can add `TODO: ` prefix anyway, so they can confirm the changes.

File: app/styles/_variables.scss [hunk 3]
```diff
@@ -7,6 +7,13 @@ $c-green-10: #d3e3e4;
 
 $c-light-blue: #8ccff4;
 $c-orange: #dd931d;
+$c-white: #fff;
+
+$c-grey-100: #272929;
+$c-grey-80: #4a4d4d;
+$c-grey-60: #7d8181;
+$c-grey-40: #b5bbbb;
+$c-grey-20: #dee4e4;
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/styles/_variables.scss:16]: yeah, it's a leftover after removing tag element. Thx for catching this up

File: app/components/ObservationCard.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import { Card, Elevation, Tooltip } from '@blueprintjs/core';
+import path from 'path';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import CreatableSelect from 'react-select/creatable';
+import classNames from 'classnames/bind';
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationCard.tsx:6]: fixed

File: app/assets/translations.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jchojna) [reply app/assets/translations.json:184]: actually adding TODO: prefix makes the label in UI looking strange, so I'll leave it as it is. It's visible in UI, so the client can suggest some change during the review if it's necessarry

File: app/components/ObservationCard.module.scss [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+          }
+        }
+      }
+
+      &:hover {
+        .data {
+          opacity: 1;
+          transform: translateX(0);
+          transition: opacity var.$t-obs-card,
+            transform var.$t-obs-card cubic-bezier(0.2, 0.98, 0.59, 1), visibility 0s;
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationCard.module.scss:133]: I change this transition to be less distractive, and added some subtle delay, to mitigate undesired effect when hovering quickly over the grid of cards

File: app/containers/ExplorePage.tsx [hunk 6]
```diff
@@ -234,10 +234,11 @@ export default function ExplorePage() {
         <Card className={styles.card} elevation={Elevation.TWO}>
           <Callout intent={Intent.PRIMARY}>{t('explore.mapHint')}</Callout>
           <div className={styles.cardBody}>
-            <Map observations={filteredObservations} onInspect={setInspectedObservations} />
+            <Map observations={filteredObservations} onInspect={() => setMapVisible(false)} />
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/containers/ExplorePage.tsx:237]: I added a separate issue to the backlog for that

File: app/styles/_variables.scss [hunk 7]
```diff
@@ -21,3 +28,5 @@ $radius-card: 10px;
 // Transitions
 $t-marker: 0.2s;
 $t-marker-animation: 1.5s;
+$t-button: 0.2s;
+$t-obs-card: 0.5s;
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/styles/_variables.scss:32]: thx

File: app/components/ObservationsInspector.tsx [hunk 8]
```diff
@@ -1,166 +1,77 @@
...
   const { t } = useTranslation();
-  if (observations.length === 0) {
-    return null;
-  }
+  const [maximizedCardIndex, setMaximizedCardIndex] = useState<number>(-1);
+  const [isMaximizedMode, setMaximizedMode] = useState<boolean>(false);
+  const toggleMaximizedMode = (cardIndex: number) => {
+    setMaximizedCardIndex(cardIndex);
+    setMaximizedMode(cardIndex >= 0);
+  };
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationsInspector.tsx:27]: Thx!

File: app/components/ObservationsInspector.tsx (hunk 1 above)
Comment (jchojna) [reply app/components/ObservationsInspector.tsx:29]: agree, fixed

File: app/containers/ExplorePage.tsx [hunk 9]
```diff
@@ -234,10 +234,11 @@ export default function ExplorePage() {
...
             <ObservationsInspector
-              observations={inspectedObservations}
-              onClose={() => setInspectedObservations([])}
+              isVisible={!isMapVisible}
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/containers/ExplorePage.tsx]: not sure why we would need to condition `<ObservationsInspector />` as a prop of `<ObservationsInspector />`. Also, in such case there are some typing errors which I couldn;'t resolve.

File: app/components/ObservationCard.tsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  const [isOverriden, setOverride] = useState<boolean>(false);
+
+  const dropdownInitialValue = {
+    value: observation.pred_1,
+    label: formatAnimalClassName(observation.pred_1)
+  };
+
+  const handlePredictionOverride = (newValue: CreatableOption | null) => {
+    onPredictionOverride(observation.location, newValue);
+    setOverride(
```
Comment (jchojna) [inline app/components/ObservationCard.tsx:42]: I've added this back from previous commit, to fix react select element which didn't update correctly. Steps to reproduce:
1. Override some card
2. Using dropdown select back the original value of this overriden card
3. The override is gone but we still can see a cross icon to remove override

File: app/containers/ExplorePage.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (kamilzyla) [reply app/containers/ExplorePage.tsx]: Perhaps I wasn't clear. You don't need `isVisible` prop in this case :wink: I've pushed a commit with a change I had in mind.

File: app/components/ObservationCard.tsx (hunk 10 above)
Comment (kamilzyla) [reply app/components/ObservationCard.tsx]: Unfortunately this won't work, as the state will be lost when the card will be unrendered in the virtualized container: override a prediction, scroll photos to the bottom and back; the "clear" button will be lost.

I have pushed a commit which should resolve this.

**Review by kamilzyla — APPROVED** [review]
LGTM, great job! Merging with some minor changes.

**Pull request merged**